lisa/schema.py

```diff
@@ -380,10 +381,16 @@ def _get_key(self) -> str:
 
     def _call_requirement_method(self, method_name: str, capability: Any) -> Any:
         # default FeatureSetting is a place holder, nothing to do.
-        extended_schemas = None
-        if hasattr(capability, "extended_schemas"):
-            extended_schemas = capability.extended_schemas
-        return FeatureSettings.create(self.type, extended_schemas=extended_schemas)
+        value = FeatureSettings.create(self.type)
+
+        # intersect the extended schemas
+        if method_name == "intersect" and capability is not None:
+            value.extended_schemas = deep_update_dict(
```

Comment:
Maybe it still needs to check if `extended_schemas` attribute exists or not. Please test it.

Reply:
Didn't find any issues as such but I have added a check to be on a safe side